var/spack/repos/builtin/packages/dlib/package.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Dlib(CMakePackage):
+    """Dlib is a modern C++ toolkit containing machine learning
+    algorithms and tools for creating complex software in C++
+    to solve real world problems."""
+
+    homepage = "http://dlib.net"
+    url      = "https://github.com/davisking/dlib/archive/v19.19.tar.gz"
+
+    version('19.19', sha256='7af455bb422d3ae5ef369c51ee64e98fa68c39435b0fa23be2e5d593a3d45b87')
+
+    variant('cuda', default=False, description='Build with CUDA')
+    variant('blas', default=True, description='Build with BLAS')
+    variant('lapack', default=True, description='Build with LAPACK')
+    variant('mkl', default=False, description='Build with the MKL library')
+    variant('avx', default=False, description='Use AVX instructions')
+    variant('shared', default=False, description='Build shared libraries')
+
+    depends_on('blas')
+    depends_on('lapack')
```

Comment:
These dependencies are not conditional on the variants. Is this correct? If so, what would be the meaning of depending on e.g. `lapack` if `~lapack`?